pkg/actuators/machine/actuator.go

```diff
@@ -612,27 +601,35 @@ func (a *Actuator) updateStatus(machine *machinev1.Machine, instance *ec2.Instan
 	} else {
 		awsStatus.InstanceID = instance.InstanceId
 		awsStatus.InstanceState = instance.State.Name
-
-		addresses, err := extractNodeAddresses(instance)
-		if err != nil {
-			glog.Errorf("%s: Error extracting instance IP addresses: %v", machine.Name, err)
-			return err
+		if instance.PublicIpAddress != nil {
+			networkAddresses = append(networkAddresses, corev1.NodeAddress{
```

Comment:
why is this dropping `extractNodeAddresses`, we want to use it here

Reply:
This was marked as resolved with no response.

I don't understand this bit. The call to `extractNodeAddresses` already sets all these. Is this needed? Wouldn't it duplicate all the node addresses?